subprojects/execution/src/main/java/org/gradle/internal/execution/caching/impl/DefaultCachingStateBuilder.java

```diff
@@ -68,9 +64,6 @@ public final void withAdditionalImplementations(Iterable<ImplementationSnapshot>
 
     @OverridingMethodsMustInvokeSuper
     protected void processAdditionalImplementation(ImplementationSnapshot additionalImplementation) {
-        if (additionalImplementation.isUnknown()) {
-            throw new GradleException("Cannot calculate build cache key for additional action with unknown implementation");
-        }
     }
```

Comment:
Nit: I wonder if these two empty implementations can be further simplified. We surely don't need the `@OverridingMethodsMustInvokeSuper` anymore.

Reply:
I removed this annotation.